### PR TITLE
posix: eventfd: depends on polling

### PR DESCRIPTION
**Pull request by galak**
Have eventfd Kconfig select POLL is the code utilizes the polling API.  We get a link error for tests/lib/fdtable/libraries.os.fdtable when building on arm-clang without this.

### CODE REVIEW

File: lib/posix/Kconfig [hunk 1]
```diff
@@ -143,6 +143,7 @@ config APP_LINK_WITH_POSIX_SUBSYS
 config EVENTFD
 	bool "Support for eventfd"
 	depends on !ARCH_POSIX
+	select POLL
```
Comment (cfriedt) [inline lib/posix/Kconfig:146]: I wonder if this is appropriate use of `imply` instead of `select`.

File: tests/posix/fs/prj.conf [hunk 2]
```diff
@@ -8,3 +8,4 @@ CONFIG_POSIX_FS=y
 CONFIG_ZTEST=y
 CONFIG_ZTEST_NEW_API=y
 CONFIG_MAIN_STACK_SIZE=4096
+CONFIG_EVENTFD=n
```
Comment (cfriedt) [inline tests/posix/fs/prj.conf:11]: Is the main reason for n-selecting this to just not link in unused code? Just curious.

Comment (galak) [reply tests/posix/fs/prj.conf:11]: That, and the test case failed on one of the arc qemu platforms w/o this.  So guessing there was a code size growth or something impact that platform.

File: lib/posix/Kconfig (hunk 1 above)
Comment (galak) [reply lib/posix/Kconfig:146]: yeah, never sure when `imply` is the right option vs `select`.  Seems like its either `depends` or `select` as the eventfd code needs `POLL` to function.

Comment (cfriedt) [reply lib/posix/Kconfig:146]: Probably `select` in this case then.

**Review by cfriedt — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**